pkgs/test_core/lib/src/runner/engine.dart

```diff
@@ -320,7 +322,13 @@ class Engine {
       }
 
       if (!_closed && setUpAllSucceeded) {
-        for (var entry in group.entries) {
+        // shuffle the group entries
+        var entries = List<GroupEntry>.from(group.entries);
```

Comment:
[nit] I don't think we want to use the `.from` constructor. This can be any of `group.entries.toList()`, `List.of(group.entries)`, or `[...group.entries]`. The `group.entries.toList()` is probably the most idiomatic choice.